internal/mock.go

```diff
@@ -0,0 +1,97 @@
+// Copyright 2018-2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"). You may
+// not use this file except in compliance with the License. A copy of the
+// License is located at
+//
+//	http://aws.amazon.com/apache2.0/
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+package internal
+
+import (
+	"context"
+
+	ptypes "github.com/gogo/protobuf/types"
+
+	"github.com/containerd/containerd/runtime/v2/task"
+)
+
+var (
+	MockTaskService task.TaskService = &mockTaskService{}
+)
+
+type mockTaskService struct {
```

Comment:
1. Can you rename it to `*_mock_test.go` to exclude this from regular builds.
2. Is there any reason not to use `mockgen` instead? ([example](https://github.com/firecracker-microvm/firecracker-containerd/blob/master/firecracker-control/local_test.go#L14))

Reply:
1. Yes
2. The only reason was I had no idea it existed, that's awesome! Will update

Reply:
Once looking into this more I'm not sure how to have mockgen generate a mock for an interface defined in a different project (task.TaskService is defined in containerd).

I also realized that right now an even simpler option is to just make this struct in task_test.go:
```
type mockTaskService struct{
    task.TaskService
}
```

so I'll go with that for now.